docs/source/virtual_rainforest/core/data.md

```diff
@@ -0,0 +1,221 @@
+{class}`~virtual_rainforest.core.data.Data` class to provide access to the different
+forcing and internal variables used in the simulation. As they are loaded, all variables
+are validated and then  added to an {class}`xarray.Dataset` object, which provides a
+consistent indexing and data manipulation for the underlying arrays of data.
+
+In many cases, a user will simply provide a configuration file to set up the data that
+will be validated and loaded when a simulation runs, but the main functionality for
+working with data using Python are shown below.
+
+## Validation
+
+One of the main functions of the {mod}`~virtual_rainforest.core.data` module is to
+automatically validate data before it is added to the `Data` instance. Validation is
+applied along a set of **core axes** used in the simulation. For a given core axis:
+
+* The dimension names of a dataset are used to identify if data should be validated on
+  that axis. For example, a dataset with `x` and `y` dimensions will be validated
+  on the `spatial` core axis.
+
+* The axis will have a set of defined validators, which are provided to handles different
+  possible data configurations. For example, there is a specific `spatial` validator
+  used to handle a dataset with `x` and `y` dimensions but no coordinate values.
+
+* When a dataset is checked against a core axis, the validation checks to see that one
+  of those validators applies to the actual configuration of the data, and then runs the
+  specific validation for that configuration.
+
+The validation process is primarily intended to check that the sizes or coordinates of
+the dimensions of provided datasets are congruent with the configuration of a particular
+simulation. Validators may also standardise or subset input datasets to map them onto a
+particular axis configuration.
+
+For more details on the different core axes and the alternative mappings applied by
+validators see the [core axis](axes.md) documentation.
+
+## Creating a `Data` instance
+
+A  {class}`~virtual_rainforest.core.data.Data` instance is created using information
+that provides information on the core configuration of the simulation. At present, this
+is just the spatial grid being used.
+
+```{code-cell}
+from pathlib import Path
+
+import tomli
+import numpy as np
+from xarray import DataArray
+
+from virtual_rainforest.core.grid import Grid
+from virtual_rainforest.core.data import Data
+from virtual_rainforest.core.axes import *
+from virtual_rainforest.core.readers import load_to_dataarray
+
+# Create a simple default grid and a Data instance
+grid = Grid()
+data = Data(grid=grid)
+
+data
+```
+
+## Adding data to a Data instance
+
+Data can be added to a {class}`~virtual_rainforest.core.data.Data` instance using one of
+two methods:
+
+1. An existing DataArray object can be added to a
+   {class}`~virtual_rainforest.core.data.Data` instance just using the standard
+   dictionary assignment:`data['var_name'] = data_array`
+
```

Comment:
The space here renders a bit weirdly imo. Not sure whether there's an easy fix though